ElrondNetwork/elrond-go
# Fix bad tx behavior through tx pool cleaning

* Refactored some misleading methods name compared with their behavior in TransactionCoordinator and PreProcess interface/implementations
* Fixed bad tx behavior through tx pool cleaning for each processed tx at commit time

**How to be tested:**

We should start an #allin system test with an additional log level set to: preprocess:TRACE
1) After a while we should verify the tx pools and they should remain almost empty. 
2) Another check should be done if already processed txs in previous block, would not be delivered as good to be processed in the next block to the next proposer by the SortedTransactionsProvider component. **(grep after "bad tx")**
